### PR TITLE
[4] Remove more deprecated code slated for removal in Joomla 4

### PR DESCRIPTION
**Pull request by PhilETaylor**
Code review

Like the code said "As of 3.7.0, passing a null value for the first argument of %1$s() is deprecated and will not be supported in 4.0."

This PR cleans up and enforces this deprecation and fixes the docblocks

`$jsSafe` is slated as a bool, but the code has code to use it as an array, but its never used as an array by Joomla core so I have not type hinted it just yet, to allow 3PD calling it

Now if you call `Text::script();` you will correctly get 

```
Too few arguments to function Joomla\CMS\Language\Text::script(), 0 passed
```

### CODE REVIEW

File: libraries/src/Language/Text.php [hunk 1]
```diff
@@ -338,30 +338,16 @@ public static function printf($string)
...
+	 * @return  array
 	 *
 	 * @since   1.7.0
 	 */
-	public static function script($string = null, $jsSafe = false, $interpretBackSlashes = true)
+	public static function script(string $string, $jsSafe = false, bool $interpretBackSlashes = true): array
 	{
-		if ($string === null)
-		{
-			Log::add(
```
Comment (Quy) [inline libraries/src/Language/Text.php]: Please remove `use Joomla\CMS\Log\Log;`.

Comment (PhilETaylor) [reply libraries/src/Language/Text.php]: Done thanks.

**Comment by RickR2H** [comment]
Hey @PhilETaylor How can we test this? I know writing testing instructions isn't fun, but it will insure not so skilled people can also test and PR's get committed faster.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34077">issues.joomla.org/tracker/joomla-cms/34077</a>.</sub>

**Comment by PhilETaylor** [comment]
> Hey @PhilETaylor How can we test this?

Code review

Also try calling `Text::script();` (with no arguments) before and after applying the PR and compare the difference.

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 416d84729c18d0ea5d673dbe9dd20370630a8786<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34077">issues.joomla.org/tracker/joomla-cms/34077</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 416d84729c18d0ea5d673dbe9dd20370630a8786<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34077">issues.joomla.org/tracker/joomla-cms/34077</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34077">issues.joomla.org/tracker/joomla-cms/34077</a>.</sub>

**Comment by brianteeman** [comment]
Could you please resolve the conflicts

**Comment by PhilETaylor** [comment]
Conflicts Resolved.

**Pull request closed without merging**